addons/survey/models/survey_user.py

```diff
@@ -383,13 +422,54 @@ def _check_answer_type_skipped(self):
     @api.model_create_multi
     def create(self, vals_list):
         for vals in vals_list:
-            suggested_answer_id = vals.get('suggested_answer_id')
-            if suggested_answer_id:
-                vals.update({'answer_score': self.env['survey.question.answer'].browse(int(suggested_answer_id)).answer_score})
+            score = self._get_answer_score(vals.get('user_input_id'), vals.get('suggested_answer_id'))
+            if score and not vals.get('answer_score'):
+                vals.update({'answer_score': score})
         return super(SurveyUserInputLine, self).create(vals_list)
 
     def write(self, vals):
-        suggested_answer_id = vals.get('suggested_answer_id')
-        if suggested_answer_id:
-            vals.update({'answer_score': self.env['survey.question.answer'].browse(int(suggested_answer_id)).answer_score})
+        score = self._get_answer_score(
+            vals.get('user_input_id'),
+            vals.get('suggested_answer_id'),
+            compute_speed_score=False)
+        if score and not vals.get('answer_score'):
+            vals.update({'answer_score': score})
         return super(SurveyUserInputLine, self).write(vals)
+
+    @api.model
+    def _get_answer_score(self, user_input_id, suggested_answer_id, compute_speed_score=True):
+        """ If score depends on the speed of the answer, we need to compute it.
+        If the user answers in less than 2 seconds, he gets 100% of the points.
+        If he answers after that, he gets minimum 50% of the points.
+        The 50 other % are ponderated between the time limit and the time it took him to answer.
+
+        If the score does not depend on the speed, we just return the answer_score field of the
+        suggested survey.question.answer. """
+
+        if not suggested_answer_id:
+            return None
+
+        answer = self.env['survey.question.answer'].search([('id', '=', suggested_answer_id)], limit=1)
+        answer_score = answer.answer_score
+
+        user_input = False
```

Comment:
This variable is not used outside the scope of the herebelow if, could be removed.